samtranslator/model/sam_resources.py

```diff
@@ -2064,3 +2065,40 @@ def _add_connector_metadata(
             except KeyError:
                 original_metadata = {}
             resource.set_resource_attribute("Metadata", {**original_metadata, **metadata})
+
+
+class SamGraphQLApi(SamResourceMacro):
+    """SAM GraphQL API Macro (WIP)."""
+
+    resource_type = "AWS::Serverless::GraphQLApi"
+    property_types = {
+        "Name": PropertyType(False, IS_STR),
+        "Tags": PropertyType(False, list_of(IS_DICT)),
+        "XrayEnabled": PropertyType(False, is_type(bool)),
+        "Auth": PropertyType(True, IS_DICT),
+    }
+
+    Auth: Auth
+    Tags: Optional[PassThrough]
+    XrayEnabled: Optional[PassThrough]
+    Name: Optional[str]
+
+    @cw_timer
+    def to_cloudformation(self, **kwargs: Any) -> List[Resource]:  # type: ignore
+        appsync_api = self._construct_appsync_api()
+        resources: List[Resource] = [appsync_api]
+
+        return resources
+
+    def _construct_appsync_api(self) -> GraphQLApi:
+        api = GraphQLApi(logical_id=self.logical_id, depends_on=self.depends_on, attributes=self.resource_attributes)
+
+        if "Type" not in self.Auth:
```

Comment:
We need more resilient solution. We don't have mandatory vs optional fields in the TypedDict, do we?

Reply:
After some offline discussion, we decided that we will remove these type checks and create a custom validator to have a more robust solution for checking valid properties are inputted by the user. This will be done in an upcoming PR.